stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/internal/TextAndAttachmentsViewHolder.kt

```diff
@@ -97,22 +88,18 @@ internal class TextAndAttachmentsViewHolder(
     }
 
     private fun setupUploads(data: MessageListItem.MessageItem) {
-        val uploadIdList: List<String> = data.message.attachments
-            .filter { attachment -> attachment.uploadState == Attachment.UploadState.InProgress }
-            .mapNotNull(Attachment::uploadId)
-
-        val needUpload = uploadIdList.isNotEmpty()
-
-        if (needUpload) {
-            clearScope()
-            val scope = CoroutineScope(DispatcherProvider.Main)
-            this.scope = scope
-
-            scope.launch {
-                trackFilesSent(context, uploadIdList, binding.sentFiles)
-            }
-        } else {
+        val totalAttachmentsCount = data.message.attachments.size
+        var completedAttachmentsCount =
+            data.message.attachments.filter { it.uploadState == null || it.uploadState == Attachment.UploadState.Success }.size
```

Comment:
Small simplification

```suggestion
            data.message.attachments.count { it.uploadState == null || it.uploadState == Attachment.UploadState.Success }
```